controllers/register.go

```diff
@@ -0,0 +1,65 @@
+package controllers
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/go-playground/validator/v10"
+	errorHandler "github.com/gutakk/go-google-scraper/helpers/error_handler"
+	session "github.com/gutakk/go-google-scraper/helpers/session"
+	"github.com/gutakk/go-google-scraper/models"
+)
+
+type RegisterController struct{}
+
+type RegisterForm struct {
+	Email           string `form:"email" binding:"email,required"`
+	Password        string `form:"password" binding:"required,min=6"`
+	ConfirmPassword string `form:"confirm-password" binding:"eqfield=Password,required"`
+}
+
+func (r *RegisterController) applyRoutes(engine *gin.Engine) {
+	engine.GET("/register", r.displayRegister)
+	engine.POST("/register", r.register)
+}
+
+func (r *RegisterController) displayRegister(c *gin.Context) {
+	c.HTML(http.StatusOK, "register.html", gin.H{
+		"title": "Register",
+	})
+}
+
+func (r *RegisterController) register(c *gin.Context) {
+	form := &RegisterForm{}
+
+	if err := c.ShouldBind(form); err != nil {
+		for _, fieldErr := range err.(validator.ValidationErrors) {
+			c.HTML(http.StatusBadRequest, "register.html", gin.H{
+				"title": "Register",
+				"error": errorHandler.ValidationErrorToText(fieldErr),
+				"email": form.Email,
+			})
+			return
+		}
+	}
+
+	hashedPassword, hashError := models.HashPassword(form.Password)
```

Comment:
Can this method really fail? If not, I would recommend moving that logic to the models or another object (e.g. user or service object). The controller should not worry about this kind of implementation details. 

As I explained in a previous comment, for this project, I think we can just move most logic to the model. So `models.SaveUser` could also call `models.HashPassword`. On large projects, it would better to have service or use case objects e.g. `services/create_user.go` or `use_cases/registration/create_user.go` which then can handle complex business logic. What we want really in the controller is just to return success/error code and messages

Reply:
This is bcrypt `GenerateFromPassword` implementation
```go
func GenerateFromPassword(password []byte, cost int) ([]byte, error) {
	p, err := newFromPassword(password, cost)
	if err != nil {
		return nil, err
	}
	return p.Hash(), nil
}
```
it returns error but I'm not sure if this case could fail.
What do you think?

Reply:
Unlikely. But your call on this to keep the error checking or not. In any case, I don't think that password hashing should be the responsibility of the controller so calling it inside `SaveUser` makes more sense. Then the method `SaveUser` could return the hashing error if it happens.

Reply:
Fixed in 659be4d2cb42d2236ac4eb36e9ba9824943b0a69.

I didn't check hash password error because I think it's ok to let `Create` function handle the error instead.